pkgs/applications/blockchains/solana/default.nix

```
@@ -0,0 +1,60 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, rustPlatform
+, IOKit
+, Security
+, AppKit
+, openssl
+, pkg-config
+, rustfmt
+, udev
+, zlib
+, llvm
+, llvmPackages
+, rocksdb
+, protobuf
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "solana-testnet-cli";
+  version = "1.9.2";
+
+  src = fetchFromGitHub {
+    owner = "solana-labs";
+    repo = "solana";
+    rev = "v${version}";
+    sha256 = "sha256-wrv35vBohLztMZPb6gfZdCaXcjj/Y7vnQqINaI6dBM4=";
+  };
+
+  cargoSha256 = "sha256-A5uVa+cRmrkVyw7MFH4QAr0VIFi18wcc2VPFvQyT9EM=";
+
+  buildAndTestSubdir = "cli";
+
+  nativeBuildInputs = lib.optionals stdenv.isLinux [ protobuf ];
+  buildInputs = lib.optionals stdenv.isLinux [ pkg-config udev zlib ] ++ lib.optionals stdenv.isDarwin [ IOKit Security AppKit ];
```

```suggestion
  nativeBuildInputs = lib.optionals stdenv.isLinux [ protobuf pkg-config ];
  buildInputs = lib.optionals stdenv.isLinux [ udev zlib ] ++ lib.optionals stdenv.isDarwin [ IOKit Security AppKit ];
```

It now fails on
```
error: failed to run custom build command for `hidapi v1.3.0`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/hidapi-aa6c9678cc52e062/build-script-build` (exit status: 101)
(...)
  thread 'main' panicked at 'Unable to find libudev: Could not run `"pkg-config" "--libs" "--cflags" "libudev"`
  The pkg-config command could not be found.

  Most likely, you need to install a pkg-config package for your OS.
  Try `apt install pkg-config`, or `yum install pkg-config`,
  or `pkg install pkg-config` depending on your distribution.

  If you've already installed it, ensure the pkg-config command is one of the
  directories in the PATH environment variable.
```

Moving `pkg-config` from `buildInputs` to `nativeBuildInputs` makes it compile.